src/Instrumentation/Laravel/src/Hooks/Queue.php

```diff
@@ -19,9 +21,30 @@ class Queue extends AbstractHook
 {
     public function instrument(): void
     {
+        $this->hookAbstractQueueCreatePayloadArray();
         $this->hookQueuePushRaw();
     }
 
+    protected function hookAbstractQueueCreatePayloadArray(): bool
+    {
+        // @todo: remove once post-hook return value works.
+        AbstractQueue::createPayloadUsing(function () {
+            $carrier = [];
+            TraceContextPropagator::getInstance()->inject($carrier);
+
+            return $carrier;
+        });
+
+        return hook(
+            AbstractQueue::class,
+            'createPayloadArray',
+            post: function (AbstractQueue $queue, array $params, array $payload, ?Throwable $exception) {
```

Comment:
```suggestion
            post: function (AbstractQueue $queue, array $params, array $payload, ?Throwable $exception): array {
```
I think this will resolve your todo above. Modified return value is ignored without a type-hint.